firmware/thruna/BMS/Inc/App/App_Eeprom.h

```diff
@@ -63,21 +65,36 @@ EEPROM_StatusTypeDef
 EEPROM_StatusTypeDef App_Eeprom_PageErase(struct Eeprom *eeprom, uint16_t page);
 
 /**
- * Write an address 4 times to EEPROM
+ * Update address and write 4 times to EEPROM
  * @note SHOULD ONLY BE CALLED ONCE EVERY 5ms, DOING SO MORE QUICKLY WILL VIOLATE EEPROM SPECS
  * @param eeprom Eeprom to write address to
- * @param page the number of the page to write to
- * @param address the address to store on the EEPROM
+ * @param address pointer to the the address to store on the EEPROM
  * @return EEPROM_StatusTypeDef returns success status for debug
  */
-EEPROM_StatusTypeDef App_Eeprom_Write4CopiesOfAddress(struct Eeprom *eeprom, uint16_t page, uint16_t address);
+EEPROM_StatusTypeDef App_Eeprom_UpdateSavedAddress(struct Eeprom *eeprom, uint16_t *address);
 
 /**
- * Read the stored 4 copies of adress from EEPROM
+ * Read stored copies of last SOC address and perform error checking
  * @param eeprom Eeprom to erase a page from
- * @param page the number of the page to read from
- * @param addresses array of 4 addresses to be chosen from. Should all contain same value but may not due to
- * data-corruption in EEPROM
- * @return EEPROM_StatusTypeDef returns success status for debug
+ * @param address address where stored SOC values can be found
+ * @return ExitCode returns EXIT_CODE_ERROR in case of data corruption or read error
+ */
+ExitCode App_Eeprom_ReadAddress(struct Eeprom *eeprom, uint16_t *address);
+
+/**
+ * Write SOC value of cell with lowest voltage to EEPROM (writes 4 identical copies to protect against data corruption)
+ * @param eeprom EEPROM to write SOC to
+ * @param min_soc value of current min_soc to write
+ * @param address page on EEPROM to store soc
+ * @return EEPROM_StatusTypeDef returns status of write operation
+ */
+EEPROM_StatusTypeDef App_Eeprom_WriteMinSoc(struct Eeprom *eeprom, float min_soc, uint16_t address);
+
+/**
+ * Read previous SOC value of cell with lowest voltage to EEPROM and perform error checking
+ * @param eeprom
+ * @param address
+ * @param min_soc
+ * @return ExitCode returns EXIT_CODE_ERROR in case of data corruption or read error
  */
-EEPROM_StatusTypeDef App_Eeprom_Read4CopiesOfAddress(struct Eeprom *eeprom, uint16_t page, uint16_t *addresses);
+ExitCode App_Eeprom_ReadMinSoc(struct Eeprom *eeprom, uint16_t address, float *min_soc);
```

Comment:
This module could probably use a better name. By name it sounds like an application-level wrapper around the EEPROM, but it seems highly specific to SOC. Consider renaming to something more accurate.

Reply:
I guess one thing I could do is rather than include the EEPROM object in the world, have it be a part of the SocStats struct as it's the only thing using the EEPROM then I could place these SOC-related EEPROM functions in App_Soc.c